src/components/login/Login.js

```diff
@@ -0,0 +1,71 @@
+import React, {useState} from 'react'
+import Logo from '../common/Logo'
+import firebase from '../../config/firebase/firebase-config'
+import 'firebase/firestore'
+import '../../assets/styles/index.css';
+import { useToasts } from 'react-toast-notifications';
+
+export function Login({setIslogin, setEmployee, setRole}) {
+    const { addToast } = useToasts()  
+    const[username, setUsername] = useState("");
+    const[password, setPassword] = useState("");
+          
+    let autenticar=(username, role)=>{
+      setIslogin(true);
+      setEmployee(username);
+      setRole(role);
+    }
+     
+    function enter(){
+      let isUser;
+      const fetchData = async () => {
+        const db = firebase.firestore();
+        const data = await db.collection('users').get()
+        data.docs.map(doc => {
+          if (doc.data().user === username.toLowerCase() && doc.data().password === password ){
+            const role = doc.data().role
+            autenticar(doc.data().name, role)
+            isUser = true;
+          }  
+        })
+
+        if(isUser){
+          return
+        }else{
+          addToast('Usuario o password incorrecto', { 
+            autoDismiss: true, 
+            placement:'top-center', 
+            appearance: 'error' })
+        }
+      }
+      fetchData();
+    }
```

Comment:
```suggestion
   function enter() {
      const db = firebase.firestore();
      // Es una buena idea filtrar el usuario desde nuestro consulta
      const data = await db.collection(`users`)
        .where('user', '==', username.toLowerCase())
        .get()
      const [foundUser] = data.docs[0]
      
      // Por qué comparamos el password?, 
      // Sugerencia: NO deberíamos guardar el password dentro de la collección users
      const role = doc.data().role
      autenticar(doc.data().name, role)
      addToast('Usuario o password incorrecto', { 
        autoDismiss: true, 
        placement:'top-center', 
        appearance: 'error' })
      })          
    }
```
Creo que no hemos logrado implementar la autenticación por medio de [Firebase Auth](https://firebase.google.com/docs/auth)

Reply:
Ya he implementado la autentificación de Firebase Auth en el proyecto de red social. En este caso no lo usé porque no es lo que estaba buscando. No quería que cualquiera pudiera registrarse, lo que quería hacer es que un admin generara un registro de empleados con contraseña para que solo él permitira quien puede y quien no usar la aplicación en una especie de control interno. Lo deje en el front porque no supe cómo lograr esto en el back con firebase. ¿Auth permite hacer esto sin que deban registrarse con un correo?

Reply:
> Ya he implementado la autentificación de Firebase Auth en el proyecto de red social. En este caso no lo usé porque no es lo que estaba buscando. No quería que cualquiera pudiera registrarse, lo que quería hacer es que un admin generara un registro de empleados con contraseña para que solo él permitira quien puede y quien no usar la aplicación en una especie de control interno. Lo deje en el front porque no supe cómo lograr esto en el back con firebase. ¿Auth permite hacer esto sin que deban registrarse con un correo?

Firebase Auth te permite crear usuarios predefinidos con solo un email y password

![dsdsdsdsdsds](https://user-images.githubusercontent.com/9435850/90432141-bc297480-e08f-11ea-9104-e20a20389eab.png)

Reply:
¡Genial, gracias!